src/lib/common/DraggablePin.svelte

```diff
@@ -0,0 +1,28 @@
+<script lang="ts">
+  import { Marker, type LngLat } from "maplibre-gl";
+  import { map } from "stores";
+  import { onDestroy, onMount } from "svelte";
+
+  let marker: Marker | null = null;
+  export let markerPosition: LngLat;
+  export let markerPositionUpdated: () => any;
+
+  onMount(() => {
+    marker = new Marker({ draggable: true })
+      .setLngLat(markerPosition)
+      .addTo($map);
+    marker.on("dragstart", () => {
+      $map.getCanvas().style.cursor = "grabbing";
+    });
+    marker.on("drag", () => {
+      $map.getCanvas().style.cursor = "inherit";
```

Comment:
Ah, actually we need to do this on `dragend` to reset it. `drag` is continuously called during drag